### PR TITLE
Set state for session instead of deleting it when deleting an applica…

### PR DESCRIPTION
**Pull request by hmusum**
…tion

Simplify by just setting the state of the active session belonging to an
application that is deleted. This is more in line with what we do for other
transitions.

Inactive sessions will be deleted by maintainers in config server after
some time.

### CODE REVIEW

File: configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionStateWatcher.java [hunk 1]
```diff
@@ -48,6 +48,7 @@ private synchronized void sessionStatusChanged(Status newStatus) {
         switch (newStatus) {
             case NEW:
             case NONE:
+            case DELETE:
```
Comment (jonmv) [inline configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionStateWatcher.java:51]: Doing nothing on DELETE means we must be sure the DEACTIVATE has already been acted upon, and the session deactivated. How can we be sure this is the case here?

**Comment by jonmv** [comment]
How long are inactive sessions kept? Do we need more aggressive GC to keep disk usage down?

File: configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionStateWatcher.java (hunk 1 above)
Comment (hmusum) [reply configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionStateWatcher.java:51]: Thanks for reminding me, that was then one thing I was thinking "that needs to be done as well", but had forgotten about today.

**Comment by hmusum** [comment]
Deleting an application is not done very often, whereas periodic redeployments are done often, so I don't think GC will need to change due to this

**Comment by jonmv** [comment]
It's the opposite for CD systems, but I guess you're right anyway.

File: configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionStateWatcher.java (hunk 1 above)
Comment (hmusum) [reply configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionStateWatcher.java:52]: Fixed

**Comment by jonmv** [comment]
👍

**Pull request merged**